src/filters/H5Zzstd.jl

```diff
@@ -0,0 +1,124 @@
+#=
+Derived from https://github.com/aparamon/HDF5Plugin-Zstandard, zstd_h5plugin.c
+Originally licensed under Apache License Version 2.0
+See H5Zzstd_LICENSE.txt
+
+The following copyright and license applies to the Julia port itself.
+Copyright © 2021 Mark Kittisopikul, Howard Hughes Medical Institute
+Licensed under MIT License, see LICENSE
```

Comment:
Let's see if I understand this. So the original license is Apache, and your port is a derivative work.
So you're licensing your contribution under your own license as opposed to using the existing HDF5.jl license? Is that normal? I guess most of our PRs are contributions and are assumed to be under the HDF5.jl license.

Reply:
Perhaps there would be a separate folder for the licenses since it's confused with source files?

Reply:
Part of this is because this code originally was from an independent package, HDF5Plugins.jl where I attached my employer's default 3-clause BSD license:
https://www.janelia.org/open-science/software-licensing

With my contribution, we have some flexibility, and I can relicense under MIT on behalf of HHMI. It should be sufficient to add `Howard Hughes Medical Institute` and me to the main LICENSE file.

Since the plugins are a derivative work, I think we just need to be very clear about what is licensed under what license. What I did over in HDF5Plugins.jl is just add the licenses to the LICENSE file:
https://github.com/JuliaIO/HDF5Plugins.jl/blob/main/LICENSE